### PR TITLE
[Concurrency] Adjustments to withTaskExecutorPreference

### PR DESCRIPTION
**Pull request by ktoso**
Sendability and typed throws for the new withTaskExecutorPreference API

Resolves rdar://122301216
Resolves rdar://122301831

### CODE REVIEW

File: stdlib/public/Concurrency/Task+TaskExecutor.swift [hunk 1]
```diff
@@ -128,12 +128,12 @@ import Swift
 /// - Throws: if the operation closure throws
 /// - SeeAlso: ``TaskExecutor``
 @_unavailableInEmbedded
-@available(SwiftStdlib 9999, *)
+@available(SwiftStdlib 6.0, *)
 @_unsafeInheritExecutor // calling withTaskExecutor MUST NOT perform the "usual" hop to global
-public func withTaskExecutorPreference<T: Sendable>(
+public func withTaskExecutorPreference<T: Sendable, E>(
```
Comment (FranzBusch) [inline stdlib/public/Concurrency/Task+TaskExecutor.swift:133]: T doesn't need to be Sendable here

Comment (ktoso) [reply stdlib/public/Concurrency/Task+TaskExecutor.swift:133]: Whoops thanks :)

Comment (ktoso) [reply stdlib/public/Concurrency/Task+TaskExecutor.swift]: Heh turns out there's all kinds of issues this causes... I don't think this change can land yet, will discuss with the team tomorrow

File: stdlib/public/Concurrency/Task+TaskExecutor.swift [hunk 2]
```diff
@@ -128,12 +128,45 @@ import Swift
 /// - Throws: if the operation closure throws
 /// - SeeAlso: ``TaskExecutor``
 @_unavailableInEmbedded
-@available(SwiftStdlib 9999, *)
+@available(SwiftStdlib 6.0, *)
+public func withTaskExecutorPreference<T, Failure>(
+  _ taskExecutor: (any TaskExecutor)?,
+  isolatedTo isolation: (any Actor)? = #isolation,
```
Comment (ktoso) [inline stdlib/public/Concurrency/Task+TaskExecutor.swift:134]: TODO: the `= #isolation` piece won't work yet

File: stdlib/public/Concurrency/Task+TaskExecutor.swift [hunk 3]
```diff
@@ -128,12 +128,45 @@ import Swift
...
+  operation: () async throws(Failure) -> T
+) async throws(Failure) -> T {
+  guard let taskExecutor else {
```
Comment (ktoso) [inline stdlib/public/Concurrency/Task+TaskExecutor.swift:137]: TODO: de-duplicate; was trying to minimize moving pieces to reproduce crashes

File: stdlib/public/Concurrency/Task+TaskExecutor.swift [hunk 4]
```diff
@@ -128,12 +128,45 @@ import Swift
 /// - Throws: if the operation closure throws
 /// - SeeAlso: ``TaskExecutor``
 @_unavailableInEmbedded
-@available(SwiftStdlib 9999, *)
+@available(SwiftStdlib 6.0, *)
+public func withTaskExecutorPreference<T, Failure>(
+  _ taskExecutor: (any TaskExecutor)?,
+  isolatedTo isolation: isolated (any Actor)? = #isolation,
```
Comment (FranzBusch) [inline stdlib/public/Concurrency/Task+TaskExecutor.swift:134]:
```suggestion
  isolation actor: isolated (any Actor)? = #isolation,
```

This follows what has been done for `AsyncSequence`

```
mutating func next(isolation actor: isolated (any Actor)?) async throws(Failure) -> Element?
```

**Comment by ktoso** [comment]
@swift-ci please test

**Pull request merged**